### PR TITLE
Add touch events and modifier states

### PR DESCRIPTION
**Pull request by vivienne-w**
This adds support for the new touch screen events and modifier states introduced in gst 1.22. Tested in chromium.

### CODE REVIEW

File: plugins/Cargo.toml [hunk 1]
```diff
@@ -9,12 +9,12 @@ repository = "https://github.com/centricular/webrtcsink/"
 build = "build.rs"
 
 [dependencies]
-gst = { package = "gstreamer", version = "0.18", features = ["v1_20", "ser_de"] }
-gst-app = { package = "gstreamer-app", version = "0.18", features = ["v1_20"] }
-gst-video = { package = "gstreamer-video", version = "0.18", features = ["v1_20", "ser_de"] }
-gst-webrtc = { package = "gstreamer-webrtc", version = "0.18", features = ["v1_20"] }
-gst-sdp = { package = "gstreamer-sdp", version = "0.18", features = ["v1_20"] }
-gst-rtp = { package = "gstreamer-rtp", version = "0.18", features = ["v1_20"] }
+gst = { git="https://gitlab.freedesktop.org/gstreamer/gstreamer-rs", rev="61a3f5296c3093e9811d883256ad38bd5483b6db", package = "gstreamer", features = ["v1_22", "ser_de"] }
```
Comment (thiblahute) [inline plugins/Cargo.toml:12]: I do not think we should always depend on 1.22 but instead have it as an option of this crate.

File: www/input.js [hunk 2]
```diff
@@ -404,10 +466,10 @@ class Input {
         // Using guacamole keyboard because it has the keysym translations.
         this.keyboard = new Keyboard(window);
         this.keyboard.onkeydown = (keysym) => {
-            this.send(JSON.stringify( {"event": "KeyPress", "key": keysym}));
+            this.send(JSON.stringify( {"event": "KeyPress", "key": keysym, "modifier_state": ""}));
```
Comment (thiblahute) [inline www/input.js:469]: Add a FIXME here at least, we should be able to have modifiers info.

File: www/input.js [hunk 3]
```diff
@@ -171,42 +183,87 @@ class Input {
...
+        event.preventDefault();
     }
 
     /**
      * Handles mouse wheel events and sends them to WebRTC app.
      * @param {MouseEvent} event
      */
     _wheel(event) {
+        let masks = [];
+        if (event.altKey) mods += "alt-mask";
```
Comment (thiblahute) [inline www/input.js:255]: Factor out a function for that?

File: plugins/Cargo.toml [hunk 4]
```diff
@@ -9,12 +9,12 @@ repository = "https://github.com/centricular/webrtcsink/"
 build = "build.rs"
 
 [dependencies]
-gst = { package = "gstreamer", version = "0.18", features = ["v1_20", "ser_de"] }
-gst-app = { package = "gstreamer-app", version = "0.18", features = ["v1_20"] }
-gst-video = { package = "gstreamer-video", version = "0.18", features = ["v1_20", "ser_de"] }
-gst-webrtc = { package = "gstreamer-webrtc", version = "0.18", features = ["v1_20"] }
-gst-sdp = { package = "gstreamer-sdp", version = "0.18", features = ["v1_20"] }
-gst-rtp = { package = "gstreamer-rtp", version = "0.18", features = ["v1_20"] }
+gst = { git="https://gitlab.freedesktop.org/gstreamer/gstreamer-rs", rev="61a3f5296c3093e9811d883256ad38bd5483b6db", package = "gstreamer", features = ["v1_20", "ser_de"] }
```
Comment (MathieuDuponchelle) [inline plugins/Cargo.toml:12]: @sdroege does that seem OK to you? :)

Comment (thiblahute) [reply plugins/Cargo.toml:12]: We depend on `gstreamer-rs` main, but decided to pin a specific commit here.

Comment (sdroege) [reply plugins/Cargo.toml:12]: Well if you want to keep this up to date :shrug:

Comment (MathieuDuponchelle) [reply plugins/Cargo.toml:12]: ok thanks, I think given webrtcsink should eventually live in `gst-plugins-rs`, we probably should do what's done there and just depend on main.

Comment (sdroege) [reply plugins/Cargo.toml:12]: You probably want to update after https://gitlab.freedesktop.org/gstreamer/gstreamer-rs/-/merge_requests/1019 is merged

Comment (MathieuDuponchelle) [reply plugins/Cargo.toml]: @vivienne-w let's just use main :)

Comment (MathieuDuponchelle) [reply plugins/Cargo.toml]: (ie you can remove the `rev` altogether)

**Comment by MathieuDuponchelle** [comment]
@thiblahute apart from the build system remarks I have nothing relevant to say, can you take over review? :)

File: plugins/Cargo.toml (hunk 4 above)
Comment (thiblahute) [reply plugins/Cargo.toml]: We still want to pin in the `.lock` file I believe.

**Comment by thiblahute** [comment]
> @thiblahute apart from the build system remarks I have nothing relevant to say, can you take over review? :)

lgtm.

**Comment by MathieuDuponchelle** [comment]
thanks!

**Pull request merged**